app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,35 @@
+module Api
+  module V1
+    class ReservationsController < ApplicationController
+      def index
+        user = User.find(params[:user_id])
+        reservations = user.reservations
+
+        if reservations.any?
+          render json: reservations, status: :ok
+        else
+          render json: { message: 'User has no reservations' }, status: :ok
+        end
+      rescue ActiveRecord::RecordNotFound
+        render json: { error: 'User not found' }, status: :not_found
+      end
+
+      def create
+        reservation = Reservation.new(reservation_params)
+        if reservation.save
+          render json: reservation, status: :created
+        else
+          render json: reservation.errors, status: :unprocessable_entity
```

Comment:
Good job handling errors. This is making easier for me to know when something is wrong with the requests on the cliente side.